src/com/facebook/buck/rules/args/WorkerMacroArg.java

```diff
@@ -99,6 +103,18 @@ public String getStartupArgs() {
     return workerTool.getArgs();
   }
 
+  public Optional<String> getPersistentWorkerKey() {
+    if (workerTool.isPersistent()) {
+      return Optional.of(buildTarget.toString());
```

Comment:
Unfortunately I think outside of the parser the cell prefix is not present in the build targets because it would have to change based on which cell references the target (two cells can be using a different names for another cell or the same name for different cells). Explicitly adding the `cellPath` from the build target should work well though.